app/policies/request_policy.rb

```diff
@@ -0,0 +1,39 @@
+class RequestPolicy < ApplicationPolicy
+  class Scope < ApplicationPolicy::Scope
+    def resolve
+      return scope.all unless Insights::API::Common::RBAC::Access.enabled?
+
+      # parent request index
+      if scope.class == Class
+        return scope.where(:parent_id => nil) if admin?(scope)
+        return scope.where(:id => approver_id_list(scope.table_name)) if approver?(scope)
+
+        scope.where(:parent_id => nil, :id => owner_id_list(scope.table_name))
+      # child request index
+      else
+        return scope if admin?(scope.model)
```

Comment:
We may be able to simply `return scope`. Approver never be able to reach here because he will be rejected at parent level. Requester should be able to see all children if he passes the parent id check.